src/ContractInteract/EIP20CoGateway.js

```diff
@@ -70,6 +95,186 @@ class EIP20CoGateway {
+   * @param {string} gatewayLib Address of GatewayLib contract
+   *                 to link into the contract bytecode.
+   * @param {Object} txOptions Transaction options.
+   *
+   * @returns {Promise<EIP20CoGateway>} Promise containing the EIP20CoGateway instance that
+   *                                    has been deployed.
+   */
+  static async deploy(
+    web3,
+    valueToken,
+    utilityToken,
+    anchor,
+    bounty,
+    organization,
+    gateway,
+    burner,
+    messageBus,
+    gatewayLib,
+    txOptions,
+  ) {
+    const tx = EIP20CoGateway.deployRawTx(
+      web3,
+      valueToken,
+      utilityToken,
+      anchor,
+      bounty,
+      organization,
+      gateway,
+      burner,
+      messageBus,
+      gatewayLib,
+    );
+
+    const _txOptions = txOptions;
+    if (!_txOptions.gas) {
+      _txOptions.gas = await tx.estimateGas();
+    }
+
+    return Utils.sendTransaction(tx, _txOptions).then((txReceipt) => {
+      const address = txReceipt.contractAddress;
+      return new EIP20CoGateway(web3, address);
+    });
+  }
+
+  /**
+   * Raw transaction object for {@link EIP20CoGateway#deploy}
+   *
+   * @param {Object} web3 Web3 object.
+   * @param {string} valueToken The value token contract address.
+   * @param {string} utilityToken The utility token address that will be used for
+   *                 minting the utility token.
+   * @param {string} anchor Address of anchor to use for getting the state root of
+   *                 the auxiliary chain.
+   * @param {string} bounty The amount that facilitator will stakes to initiate the
+   *                 stake process.
+   * @param {string} organization Address of an organization contract.
+   * @param {string} gateway Address of EIP20Gateway on Origin.
+   * @param {string} burner Address where tokens will be burned.
+   * @param {string} messageBusAddress Address of MessageBus contract
+   *                 to link into the contract bytecode.
+   * @param {string} gatewayLibAddress Address of GatewayLib contract
+   *                 to link into the contract bytecode.
+   * @param {Object} txOptions Transaction options.
+   *
+   * @returns {Promise<Object>} Promise that resolves to raw transaction object.
+   */
+  static deployRawTx(
+    web3,
+    valueToken,
+    utilityToken,
+    anchor,
+    bounty,
+    organization,
+    gateway,
+    burner,
+    messageBusAddress,
+    gatewayLibAddress,
+  ) {
+    const messageBusLinkInfo = {
+      address: messageBusAddress,
+      name: 'MessageBus',
+    };
+    const gatewayLibLinkInfo = {
+      address: gatewayLibAddress,
+      name: 'GatewayLib',
+    };
+
+    const abiBinProvider = new AbiBinProvider();
+    const abi = abiBinProvider.getABI(ContractName);
+    const bin = abiBinProvider.getLinkedBIN(
+      ContractName,
+      messageBusLinkInfo,
+      gatewayLibLinkInfo,
+    );
+
+    const contract = new web3.eth.Contract(abi, null, null);
+    const args = [
```

Comment:
We *could* call `args` `arguments` everywhere 🤔

Reply:
No, we can't, since `arguments` is a JS keyword 😅

Reply:
Oops sorry 🤦‍♂️